### PR TITLE
updates so consent can micromanage its own status

### PR DESCRIPTION
**Pull request by rushtong**
No Jira.

Consent will return 200 even if subsystems fail. 
If consent is down it will error out.

Requires changes to Firecloud-Develop: https://github.com/broadinstitute/firecloud-develop/pull/985

---

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Make sure Swagger is updated if API changes
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Review by davidangb — APPROVED** [review]
:+1: did not test

File: consent-ws/src/main/java/org/broadinstitute/consent/http/resources/StatusResource.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        this.healthChecks = healthChecks;
+    }
+
+    @GET
+    @Produces("application/json")
+    public Response getStatus() {
+        Map<String, HealthCheck.Result> results = healthChecks.runHealthChecks();
+        HealthCheck.Result mysql = results.getOrDefault("mysql", HealthCheck.Result.unhealthy("Unable to access mysql database"));
+        HealthCheck.Result mongodb = results.getOrDefault("mongodb", HealthCheck.Result.unhealthy("Unable to access mongodb database"));
+        if (!mysql.isHealthy() || !mongodb.isHealthy()) {
```
Comment (aednichols) [inline consent-ws/src/main/java/org/broadinstitute/consent/http/resources/StatusResource.java]: 100% optional/pedantic - this feels like a double negative; might be clearer as
```
if (mysql.isHealthy() && mongodb.isHealthy()) {
  return Response.ok(results).build();
} else {
[...]
```

**Review by aednichols — APPROVED** [review]
LGTM 👍

File: consent-ws/src/main/java/org/broadinstitute/consent/http/resources/StatusResource.java (hunk 1 above)
Comment (rushtong) [reply consent-ws/src/main/java/org/broadinstitute/consent/http/resources/StatusResource.java]: 👍 - I tend to fall back into bad habits :-(

Comment (aednichols) [reply consent-ws/src/main/java/org/broadinstitute/consent/http/resources/StatusResource.java]: Eh, pretty superficial. I can think of worse coding habits to have 🙂

**Pull request merged**